UnityProject/Assets/Scripts/Editor/Tools/GameObjectToPrefab.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+/// <summary>
+/// Will convert a traditional GameObject and all its child objects to prefabs.
+/// </summary>
+public class GameObjectToPrefab : EditorWindow
+{
+	private string path;
+	private bool createEmptyPrefabs = false;
+	private bool createFolders = false;
+	private GameObject parentObject;
+
+	[MenuItem("Tools/GameObject to Prefab Converter")]
+	private static void Init()
+	{
+		// Get existing open window or if none, make a new one:
+		GameObjectToPrefab window = (GameObjectToPrefab)GetWindow(typeof(GameObjectToPrefab));
+		window.titleContent.text = "GameObject to Prefab Converter";
+		window.Show();
+	}
+
+	private void SelectPath()
+	{
+		string baseSTR = EditorUtility.OpenFolderPanel("Select folder to save prefabs", Application.dataPath, "prefabs");
+		if (baseSTR != "")
+		{
+			if (-1 == baseSTR.IndexOf(Application.dataPath, System.StringComparison.InvariantCultureIgnoreCase))
+			{
+				EditorUtility.DisplayDialog("Assets folder error", "Target folder must be any child directory to : \"" + Application.dataPath + "\"", "OK");
+				path = "";
+			}
+			else
+			{
+				path = baseSTR.Replace(Application.dataPath, "Assets");
+			}
+		}
+	}
+
+	private void CreatePrefabsRecursive(GameObject parentObject, string currentPath)
+	{
+		string newPath = currentPath;
+		int childCount = parentObject.transform.childCount;
+
+		if ((childCount > 0) && createFolders)
+		{
+			newPath += "/" + parentObject.name;
+			Debug.Log($"Creating folder {newPath}");
+			Directory.CreateDirectory(newPath);
+		}
+
+		Component[] components = parentObject.GetComponents<Component>();
+		bool hasComponents = !((components.Length == 1) && (components[0].GetType() == typeof(Transform)));
+		if (createEmptyPrefabs || hasComponents)
+		{
+			Debug.Log($"Saving Prefab {parentObject.name}");
+			PrefabUtility.SaveAsPrefabAsset(parentObject, $"{newPath}/{parentObject.name}.prefab");
+		}
+
+		for (int childIndex = 0; childIndex < childCount; childIndex++)
+		{
+			CreatePrefabsRecursive(parentObject.transform.GetChild(childIndex).gameObject, newPath);
+		}
+	}
+
+	void OnGUI()
+	{
+		EditorGUILayout.HelpBox("Choose the GameObject that contains the childs you want to convert to prefabs", MessageType.Info); ;
```

Comment:
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)